astrobee/survey_manager/survey_planner/src/dock_action_node.cpp

```diff
@@ -0,0 +1,128 @@
+/* Copyright (c) 2021, United States Government, as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ *
+ * All rights reserved.
+ *
+ * The "ISAAC - Integrated System for Autonomous and Adaptive Caretaking
+ * platform" software is licensed under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <ros/ros.h>
+
+#include <plansys2_executor/ActionExecutorClient.hpp>
+
+#include <sys/types.h>
+#include <sys/wait.h>
+
+#include <algorithm>
+#include <string>
+#include <iostream>
+
+namespace plansys2_actions {
+
+class DockAction : public plansys2::ActionExecutorClient {
+ public:
+  DockAction(ros::NodeHandle nh, const std::string& action, const std::chrono::nanoseconds& rate)
+      : ActionExecutorClient(nh, action, rate) {
+    progress_ = 0.0;
+    process_pipe_ = nullptr;
+  }
+
+ protected:
+  void do_work() {
+    std::string from, towards;
+
+    if (get_arguments().size() > 2) {
+      robot_name_ = get_arguments()[0];
+      from = get_arguments()[1];
+      towards = get_arguments()[2];
+    } else {
+      finish(false, 1.0, "Not enough arguments for [DOCK] command");
+    }
```

Comment:
I had been talking with Marina about this and cleaned it up right after you wrote this (new version is added to this PR). It is not quite a pure pass through, probably would have been a better idea to write these nodes in python so that would be easier as you say, but I think it's good enough for now.

Reply:
I saw some updates that might be what you're referring to and it looks much much better.

However, my next step was wondering why we need multiple node binaries at all. Can we not spawn multiple copies of an identical node and tell each one what action to register for using a roslaunch parameter? (My roslaunch familiarity is weak.) It would be nice to get rid of all of those node source files and their corresponding cmake build directives. But not a huge deal either way.

Reply:
We could, yeah, not a huge deal either way as you say though so I probably won't fix it for this PR.